sqldb/sqlc/queries/invoices.sql

```diff
@@ -169,3 +169,23 @@ INSERT INTO invoice_htlc_custom_records (
 SELECT ihcr.htlc_id, key, value
 FROM invoice_htlcs ih JOIN invoice_htlc_custom_records ihcr ON ih.id=ihcr.htlc_id 
 WHERE ih.invoice_id = $1;
+
+-- name: InsertInvoicePaymentHashAndKey :exec
+INSERT INTO invoice_payment_hashes (
+    hash, id
+) VALUES (
+    $1, $2
+);
+
+-- name: SetInvoicePaymentHashAddIndex :exec
+UPDATE invoice_payment_hashes
+SET add_index = $2
+WHERE id = $1;
+
+-- name: GetInvoicePaymentHashByAddIndex :one
+SELECT hash
+FROM invoice_payment_hashes
```

Comment:
just confirming understanding: we might have invoices with no add index right? but that is only the case for invoices that defs have preimages and so we would never need to actually call this method for those?

Reply:
AFAIK all invoices must have and add index. It's supposed purpose was/is to be able to read invoices added after a certain point (after restarts).

Reply:
ah ok i see: it is just to accommodate quickly creating this table given the structure of the KVDB? 

i see in `createInvoiceHashIndex`, we do:
1) iterate over all payment hashes -> invoice ID pairs and insert those here. 
2) iterate over `pay_addr_index` and for each pay-addr-> invoice ID, update that entry here. 

The only risk i see there is that we need to be very sure that there will always be an add index entry for all the invoices since these 2 steps are done separately like this. 

Perhaps we can be very sure of this though & that there is no problem?

If we wanted to be super safe though, we could do:

1) iterate over `payment_hashes` bkt - get each `invoice ID` -> use that to get the serialised invoice from `invoice_key` bucket -> get the `addr_index` from that deserialisation. This means you can insert the `addr_index` -> `payment_hash` directly and skip the `invoice_key` & no need for the two steps. 

but yeah - again if we are sure that the addr_index is defs consistent with the payment hash index, then should be ok to leave as is